### PR TITLE
Hausdorff metric added

### LINKED ISSUE #109: Add hausdorff 95
**Is your feature request related to a problem? Please describe.**
Hausdorff metric would be good to have.

**Describe the solution you'd like**
MedPy has an implementation to use, but using the entire library is cumbersome, so perhaps that implementation can be used.

**Describe alternatives you've considered**
Using this metric from a different package.

**Additional context**
N.A.

### PR DESCRIPTION
**Pull request by sarthakpati**
Fixes #109

## Proposed Changes

- adds hausdorff metric for segmentation
- some minor bugfixes in the training loop to make the output saving more coherent
- fixed the way some comments were written so that black wouldn't reformat
- includes #108

## Checklist

- [x] I have read the [`CONTRIBUTING`](https://github.com/CBICA/GaNDLF/blob/master/CONTRIBUTING.md) guide
- [x] My PR is based from the [current GaNDLF master ](https://garygregory.wordpress.com/2016/11/10/how-to-catch-up-my-git-fork-to-master/)
- [x] Non-breaking change (would not break existing functionality): if changes breaks current code, please provide as many details as possible
- [x] Function/class source code documentation added/updated
- [x] Code has been [blacked](https://github.com/psf/black#usage) for style consistency
- [x] If applicable, version information [has been updated in GANDLF/version.py](https://github.com/CBICA/GaNDLF/blob/master/GANDLF/version.py)
- [x] If adding a submodule, add to list of expections for black styling in [pyproject.toml](https://github.com/CBICA/GaNDLF/blob/master/pyproject.toml) file 
- [x] [Usage documentation](https://github.com/CBICA/GaNDLF/blob/master/docs) has been updated, if appropriate
- [x] [History](https://github.com/CBICA/GaNDLF/blob/master/HISTORY.md) has been updated, if appropriate
- [x] Tests added or modified to cover the changes
- [x] Integration tests passed locally by running `pytest`

### CODE REVIEW

File: GANDLF/utils.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ def reverse_one_hot(predmask_array, class_list):
     Returns:
         torch.Tensor: The final mask torch.Tensor.
     """
-    idx_argmax = np.argmax(predmask_array, axis=0)
+    idx_argmax = np.argmax(predmask_array.cpu().detach().numpy(), axis=0)
```
Comment (Geeks-Sid) [inline GANDLF/utils.py:143]: Do we have to detach from graph?

**Review by Geeks-Sid — CHANGES_REQUESTED** [review]

File: GANDLF/utils.py (hunk 1 above)
Comment (sarthakpati) [reply GANDLF/utils.py:143]: No, but it is the only thing that worked.

Comment (sarthakpati) [reply GANDLF/utils.py:143]: Done.

**Review by Geeks-Sid — APPROVED** [review]
